lib/wumpex/gateway.ex

```diff
@@ -13,10 +64,171 @@ defmodule Wumpex.Gateway do
 
   Invalid messages will cause Discord to close the gateway.
   """
-  @spec dispatch(gateway :: pid(), opcode :: map()) :: :ok
-  def dispatch(gateway, opcode) when is_pid(gateway) and is_map(opcode) do
+  @spec send_opcode(gateway :: pid(), opcode :: map()) :: :ok
+  def send_opcode(gateway, opcode) when is_pid(gateway) and is_map(opcode) do
     encoded = :erlang.term_to_binary(opcode)
 
-    Websocket.send(gateway, encoded, mode: :binary)
+    Websocket.send(gateway, {:binary, encoded})
+  end
+
+  @impl Websocket
+  def on_connected(options) do
```

Comment:
```suggestion
  @doc """
  Builds the gateway state
  """
  def on_connected(options) do
```
maybe add some docs here

Reply:
Documentation is implied by the `@impl Websocket` directive, which marks this an implementation of the websocket behaviour (which is documented).